src/components/autocomplete/autocomplete.ts

```diff
@@ -123,9 +123,10 @@ export class SbbAutocomplete extends LitElement {
       return;
     }
 
-    this._state = 'opening';
-    this._willOpen.emit();
-    this._setOverlayPosition();
+    if (this._willOpen.emit()) {
+      this._state = 'opening';
+      this._setOverlayPosition();
+    }
```

Comment:
This comment would be for all files (not only this one):
I personally would prefer exits with "return" in those cases, as then the flow (especially if there is more content in a block) is not nested. But probably you have a different opinion?

```suggestion
    if (!this._willOpen.emit()) {
      return
    }
    this._state = 'opening';
    this._setOverlayPosition();
```

Reply:
I don't have a strong preference between the two solutions. I can change it 😄